### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by deepueg**

### CODE REVIEW

**Review by asharpbflat — APPROVED** [review]

File: android/lib/src/main/java/com/ern/api/impl/core/ElectrodeBaseFragmentDelegate.java [hunk 1]
```diff
@@ -172,13 +172,13 @@ private Bundle initialProps(boolean isFragmentBeingReconstructed) {
...
     public void refresh(@Nullable Bundle data) {
-        if (mMiniAppView != null) {
+        if (mMiniAppView instanceof ReactRootView) {
             Bundle props = getDefaultProps();
             addGlobalProps(props);
             if (data != null) {
                 props.putAll(data);
             }
-            mMiniAppView.setAppProperties(props);
+            ((ReactRootView)mMiniAppView).setAppProperties(props);
```
Comment (friederbluemle) [inline android/lib/src/main/java/com/ern/api/impl/core/ElectrodeBaseFragmentDelegate.java:181]: Missing space after cast

File: android/lib/build.gradle [hunk 2]
```diff
@@ -16,6 +16,9 @@ android {
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
+        debug {
+            multiDexEnabled true
```
Comment (friederbluemle) [inline android/lib/build.gradle:20]: Is this required..?

File: android/lib/src/androidTest/java/com/ern/api/impl/BasicNavigationApiTest.java [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+    public void setUp() {
+        Logger.overrideLogLevel(Logger.LogLevel.VERBOSE);
+    }
+
+    @Rule
+    public ActivityScenarioRule<SampleActivity> rule = new ActivityScenarioRule<SampleActivity>(SampleActivity.class);
+
+    @Test
+    public void testRootComponentRendering() {
+        Assert.assertNotNull(rule);
```
Comment (friederbluemle) [inline android/lib/src/androidTest/java/com/ern/api/impl/BasicNavigationApiTest.java:60]: Best practice to use static imports for things like `assertNotNull` etc. (Just like you've already added a static import for `fail`) - It makes these lines shorter and easier to read.

**Review by friederbluemle — APPROVED** [review]
Changes look good overall. The `Upgrade gradle` commit was missing some updates to the wrapper scripts - something that the `wrapper` task does (Android Studio won't do it). I opened a quick PR with _only_ those changes here: #152
So after rebasing, you can exclude the file from the commit, and also reword it to `Update Android Gradle plugin to 4.0.1`

File: android/lib/build.gradle [hunk 4]
```diff
@@ -27,7 +30,10 @@ dependencies {
     implementation 'androidx.appcompat:appcompat:1.1.0'
     implementation 'androidx.lifecycle:lifecycle-extensions:2.1.0'
     api ('com.walmartlabs.ern:container-movies-reloaded:1.0.0') {changing=true}
-    testImplementation 'junit:junit:4.12'
-    androidTestImplementation 'androidx.test.ext:junit:1.1.1'
-    androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
+    androidTestImplementation 'androidx.swiperefreshlayout:swiperefreshlayout:1.1.0'
+    androidTestImplementation 'com.android.support:multidex:1.0.3'
```
Comment (krunalsshah) [inline android/lib/build.gradle:34]: you can use `"androidx.multidex:multidex:2.0.1"` instead

File: android/lib/build.gradle (hunk 2 above)
Comment (deepueg) [reply android/lib/build.gradle:20]: Yes, required for instrumentation testing

Comment (friederbluemle) [reply android/lib/build.gradle:20]: But this enables it not only for testing... Why is it required for instrumentation testing?
